### PR TITLE
new summary drilldown downs, testing circles plot for subpopulation b…

### LINKED ISSUE #54: add ordered bar chart of services to results page
- [ ] Add an ordered (horizontal) bar chart to the results page that breaks down which population groups contribute to the "Total 'surge' referrals" output box.
- [ ] Show the values of the bars as labels at the end of each bar

### LINKED ISSUE #56: Display of population groups at baseline
Create a visual of the population groups at baseline, colours by their higher level descriptions (page 10 of report). Suggestion of visualisation is to use a [circle packing method](https://www.r-graph-gallery.com/circle-packing.html#:~:text=Circular%20packing%20or%20circular%20treemap,as%20circles%20inside%20of%20it.)

### PR DESCRIPTION
**Pull request by Victor-GY-Yu**
#56 and #54

### CODE REVIEW

File: shiny_app/server.R [hunk 1]
```diff
@@ -351,4 +351,153 @@ shinyServer(function(input, output, session) {
...
+                                 subpopn %in% c("Learning disabilities & autism", "Pre existing CMH illness", "Pre existing LTC", "Pre existing SMI") ~ "Existing Conditions",
+                                 TRUE ~ NA_character_))
+
+
+
+    packing <- circleProgressiveLayout(circle_pack_plot$value, sizetype='area')
+    circle_pack_plot <- cbind(circle_pack_plot, packing)
+    dat.gg <- circleLayoutVertices(packing, npoints=50)
+
+    dat.gg <- dat.gg %>% left_join(tibble(level_2 = circle_pack_plot$level_2, id = 1:16),
```
Comment (tomjemmett) [inline shiny_app/server.R]: merge lines 439 and 441 into a single pipeline

**Review by tomjemmett — CHANGES_REQUESTED** [review]
fix linting issues and ready to merge

File: shiny_app/server.R [hunk 2]
```diff
@@ -351,4 +351,153 @@ shinyServer(function(input, output, session) {
...
+  output$surge_treatmentpathwayplot <- renderPlot({
+    summary_treatment_pathway() %>%
+      pivot_longer(!contains("treatment_pathway")) %>%
+      ggplot(aes(treatment_pathway, value)) +
+      geom_col(position = "dodge") +
+      labs(y = "Number",
+             x = "Treatment Pathway") +
+      scale_x_discrete(labels = function(x) str_wrap(x, 8),
+                       guide = guide_axis(n.dodge = 2)) +
+      facet_wrap(~ name, ncol=1)
```
Comment (tomjemmett) [inline shiny_app/server.R]: I would probably migrate to the more modern ggplot/tidy syntax of `facet_wrap(vars(name), ncol = 1)`

File: shiny_app/server.R [hunk 3]
```diff
@@ -351,4 +351,153 @@ shinyServer(function(input, output, session) {
...
+
+  })
+
+  ## Bubble Pack testing ####
+
+  output$bubble_plot_baselinepopn <- renderPlotly({
+
+    circle_pack_plot <- map_dbl(params$groups, "size") %>%
+      enframe(name = "subpopn") %>%
+      mutate(level_2 = case_when(subpopn %in% c("Children & young people", "Students FE & HE") ~ "Children & young people",
```
Comment (tomjemmett) [inline shiny_app/server.R]: a lookup table may be easier to work with than a case_when in this instance:
```R
params$groups %>%
      map_dbl("size") %>%
      enframe(name = "subpopn") %>%
      left_join(
        tribble(
          ~subpopn,                         ~level_2,
          "Children & young people",        "Children & young people",
          "Students FE & HE",               NA,
          "Elderly alone",                  "Elderly alone",
          "General population",             "General population",
          "Domestic abuse victims",         "Other Adults and Specific Groups",
          "Family of COVID deceased",       NA,
          "Family of ICU survivors",        NA,
          "Newly unemployed",               NA,
          "Pregnant & New Mothers",         NA,
          "Parents",                        NA,
          "Health and care workers",        "Directly affected individuals",
          "ICU survivors",                  NA,
          "Learning disabilities & autism", "Existing Conditions",
          "Pre existing CMH illness",       NA,
          "Pre existing LTC",               NA,
          "Pre existing SMI",               NA
        ) %>% fill(level_2),
        by = "subpopn"
      )
```

**Review by tomjemmett — CHANGES_REQUESTED** [review]

**Comment by tomjemmett** [comment]
@VictorNautica I think this will close #50, #51 and #52? Will link these if so

File: shiny_app/server.R (hunk 1 above)
Comment (Victor-GY-Yu) [reply shiny_app/server.R]: merged lines

File: shiny_app/server.R (hunk 3 above)
Comment (Victor-GY-Yu) [reply shiny_app/server.R]: replaced - works fine

File: shiny_app/server.R [hunk 4]
```diff
@@ -351,4 +351,204 @@ shinyServer(function(input, output, session) {
...
+
+  output$surge_subpopn <- renderTable({
+    summary_outputs() %>%
+      pivot_wider(names_from = type, values_from = value) %>%
+      dplyr::group_by(group) %>%
+      filter(!is.na(group)) %>%
+      summarise(`Adjusted exposed / at risk @ baseline` = as.integer(round(sum(`new-at-risk`, na.rm = T), 0)),
+                `Total symptomatic over period referrals` = as.integer(round(sum(`new-referral`, na.rm = T), 0)),
+                `Total receiving services over period` = as.integer(round(sum(`new-treatment`, na.rm = T), 0))) %>%
+      arrange(-`Total symptomatic over period referrals`) %>%
```
Comment (tomjemmett) [inline shiny_app/server.R]: I would generally suggest using `arrange(desc(column))` vs `arrange(-column)` due to the confusion that can be caused with `select(-col)` etc.

**Pull request merged**